unitary/alpha/quantum_world.py

```diff
@@ -72,11 +71,22 @@ def add_object(self, obj: QuantumObject):
         """
         if obj.name in self.used_object_keys:
             raise ValueError("QuantumObject {obj.name} already added to world.")
-        self.used_object_keys.add(obj.name)
-        self.objects.append(obj)
+        self.used_object_keys[obj.name] = obj
```

Comment:
this should not be called `_keys` anymore. `named_objects_dict`, maybe?

Reply:
Changed to object_name_dict.